applications/asset_tracker_v2/src/modules/location_module.c

```diff
@@ -0,0 +1,624 @@
+		struct location_module_event *event = cast_location_module_event(aeh);
+		struct location_msg_data msg = {
+			.module.location = *event
+		};
+
+		message_handler(&msg);
+	}
+
+	if (is_modem_module_event(aeh)) {
+		struct modem_module_event *event = cast_modem_module_event(aeh);
+		struct location_msg_data msg = {
+			.module.modem = *event
+		};
+
+		message_handler(&msg);
+	}
+	return false;
+}
+
+/* Static module functions. */
+static void inactivity_timeout_handler(struct k_timer *timer_id)
+{
+	/* Simulate a sleep event from GNSS with older MFWs. */
+	int event = NRF_MODEM_GNSS_EVT_SLEEP_AFTER_TIMEOUT;
+
+	k_msgq_put(&event_msgq, &event, K_NO_WAIT);
+}
+
+static K_TIMER_DEFINE(inactivity_timer, inactivity_timeout_handler, NULL);
+
+static void timeout_send(void)
+{
+	struct location_module_event *location_module_event = new_location_module_event();
+
+	location_module_event->data.location.search_time = stats.search_time;
+	location_module_event->type = LOCATION_MODULE_EVT_TIMEOUT;
+
+	APP_EVENT_SUBMIT(location_module_event);
+}
+
+static void data_send_pvt(void)
+{
+	struct location_module_event *location_module_event = new_location_module_event();
+
+	location_module_event->data.location.pvt.longitude = pvt_data.longitude;
+	location_module_event->data.location.pvt.latitude = pvt_data.latitude;
+	location_module_event->data.location.pvt.altitude = pvt_data.altitude;
+	location_module_event->data.location.pvt.accuracy = pvt_data.accuracy;
+	location_module_event->data.location.pvt.speed = pvt_data.speed;
+	location_module_event->data.location.pvt.heading = pvt_data.heading;
+	location_module_event->data.location.timestamp = k_uptime_get();
+	location_module_event->type = LOCATION_MODULE_EVT_DATA_READY;
+
+	location_module_event->data.location.satellites_tracked = stats.satellites_tracked;
+	location_module_event->data.location.search_time =
+		(uint32_t)(location_module_event->data.location.timestamp - stats.start_uptime);
+
+	APP_EVENT_SUBMIT(location_module_event);
+}
+
+static void search_start(void)
+{
+	int err;
+	struct location_config config;
+	enum location_method methods[2];
+	int methods_count = 0;
+	int methods_index_gnss = -1;
+	int methods_index_cellular = -1;
+
+	if (copy_cfg.no_data.neighbor_cell && copy_cfg.no_data.gnss) {
+		return;
+	}
+
+	if (!copy_cfg.no_data.gnss) {
+		methods[methods_count] = LOCATION_METHOD_GNSS;
+		methods_index_gnss = methods_count;
+		methods_count++;
+	}
+
+	/* If GNSS shouldn't be requested yet, just request cellular location */
+	if (!copy_cfg.no_data.neighbor_cell) {
+		methods[methods_count] = LOCATION_METHOD_CELLULAR;
+		methods_index_cellular = methods_count;
+		methods_count++;
+	}
+
+	if (methods_count == 0) {
+		LOG_WRN("No location methods selected");
+		return;
+	}
+
+	location_config_defaults_set(&config, methods_count, methods);
+
+	if (methods_index_gnss != -1) {
+		config.methods[methods_index_gnss].gnss.timeout =
+			copy_cfg.gnss_timeout * MSEC_PER_SEC;
```

Comment:
Now we need to conclude what to do with gnss_timeout. Some discussion here: https://github.com/nrfconnect/sdk-nrf/pull/8631/files#r973988723

Right now gnss_timeout from ATv2 configuration is set to location library's gnss timeout which starts when GNSS gets run time:
-  with A-GPS when LTE goes to idle
- without A-GPS when LTE goes to PSM

In the previous meeting, there was discussion we could rename gnss_timeout to location_timeout and let this be wall clock time. Location library now has location request wide timeout unlike mentioned here a month ago:
https://github.com/nrfconnect/sdk-nrf/pull/8631/files#r974169139

If we want this location request timeout rather than gnss timeout, we'd have dependency on the cloud side as that should be also changed, right?
I've tried to avoid dependencies so that we get this big PR closed at some point. If cloud becomes bottleneck, maybe we can do the renaming in the code, set this value to location request timeout (even though in cloud it's still gnss_timeout), and do the final change in codecs when cloud is ready with a separate PR.

Comments?

Reply:
@coderbyheart Correct me if I'm wrong here. This shouldn't create any large effort cloud side. Its mostly documentation and renaming. The current protocol name for this configuration is `gnsst`. I would discuss this further with Markus.

Reply:
Correct, there is not big change involved on the cloud side. Existing device shadows need to have the property `gnsst` removed so they do not create a delta on AWS, because the firmware will no longer handle it.

Reply:
I've changed gnss timeout to location timeout throughout the code. However, AWS and Azure fields are still gnsst to keep cloud working properly.

Reply:
> @coderbyheart Is it possible to update the cloud backend to accept both `gnsst` and the new value for the time being? 

Yes, that is possible. But I would let the cloud determine (based on the running firmware version) which value name to use. But if you want to keep backwards compatibility (for how long?) because you assume that the cloud side will not be updated (which I think is unlikely) then you can have the handling for both names in the code.

> I don't know what you agreed on in terms of naming `loct`?

First time I hear about it. But is fine with me.

Reply:
`loct` is what I had in the code locally before reverting back to `gnsst` so that's fine.

Reply:
`gnsst` renamed to `loct`